sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -225,17 +228,21 @@ class StreamExecution(
   }
 
   /**
-   * Queries all of the sources to see if any new data is available. When there is new data the
-   * batchId counter is incremented and a new log entry is written with the newest offsets.
-   *
-   * Note that committing the offsets for a new batch implicitly marks the previous batch as
-   * finished and thus this method should only be called when all currently available data
-   * has been written to the sink.
+   * Commit the batch. Note that committing the offsets for a new batch implicitly marks the
+   * previous batch as finished and thus this method should only be called when all currently
+   * available data has been written to the sink.
    */
-  private def commitAndConstructNextBatch(): Boolean = {
+  private def commitBatch(): Unit = {
```

Comment:
Why does this need to be a separate function anyways, and not be part of runBatch(). It gets called only after runBatch().